.github/workflows/auto-integrate-node-release.yaml

```diff
@@ -0,0 +1,238 @@
+name: Auto-Integrate Node Release
+
+on:
+  repository_dispatch:
+    types: [node-release-published]
+  workflow_dispatch:
+    inputs:
+      node_version:
+        description: "Node version (e.g., 0.18.0-rc.5)"
+        required: true
+        type: string
+      metadata_url:
+        description: "URL to metadata.scale file (optional - will try to download from release if not provided)"
+        required: false
+        type: string
+
+env:
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: full
+
+jobs:
+  integrate:
+    name: Integrate Node ${{ github.event.client_payload.version || inputs.node_version }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - name: Checkout indexer repository
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set version variable
+        id: version
+        run: |
+          VERSION="${{ github.event.client_payload.version || inputs.node_version }}"
+          echo "node_version=$VERSION" >> "$GITHUB_OUTPUT"
+          echo "Node version: $VERSION"
+
+      - name: Check if branch already exists
+        id: check_branch
+        run: |
+          BRANCH="feat/integrate-node-${{ steps.version.outputs.node_version }}"
```

Comment:
Here comes a tricky question: Is this a feature? Or a dependency update? IMHO the answer depends ... once we support multiple node versions, e.g. 0.18 and 0.19, only supporting a new node version is a feature.

Also, do we want to automatically integrate with EVERY node release?